torch/_decomp/decompositions.py

```diff
@@ -1988,9 +1988,28 @@ def upsample_compute_output_size(input_size, output_size, scale_factors):
         )
         utils.check(len(scale_factors) == spatial_dimensions, lambda: "")
         return [
-            # Returning output_size as float. We cannot convert it to int directly,
-            # as latter computation of scale_factor is relying output size being float
-            sym_float(input_size[i + 2] * scale_factors[i])
+            # XXX: The int call below is not correct since it will discard info
+            # if an FP number has non-zeros after the dot. We are only testing
+            # with 2.0, so this is a good first step towards fixing this the
+            # right way. Next, I would try comparing float and int codegen if
+            # you can get it to JIT-compile.
+            #
+            # Or we could try flooring like this instead, but it fails on CPU
+            # while the int version doesn't. We use floordiv here because it has
+            # a lowering already and works for symbols. Another option would be
+            # doing sympy.floor() since math.floor() doesn't work for symbols.
+            #
+            # ((input_size[i + 2] * scale_factors[i]) // 1)
+            #
+            # error: invalid controlling predicate
+            #    for(long i0=0; i0<(-1)*ks0*(floor(-2.0*ks1)); i0+=1)
+            #
+            # On CUDA, the int code fails due to recent changes in the triton
+            # codegen, but there are likely more problems since it used to fail
+            # before with a numerical accuracy error:
+            # torch/_inductor/codegen/triton.py", line 758, in indexing
+            # assert var.name[0] in "xyr", var.name
+            sym_float(input_size[i + 2] * int(scale_factors[i]))
```

Comment:
int is wrong here, see the comment above

Reply:
this doesn't work right now, but the proper way is `sym_floor(input_size[i + 2] * scale_factors[i])` (but it requires extending `sym_floor`). in theory, it should tell the codegen that it's a `symint` and it should work fine with `arange`, but then we need to make sure that FP precision is used again when `x` and `y` are multiplied in the decomp below.

https://pytorch.org/docs/stable/generated/torch.nn.Upsample.html (see the floor brackets in the image there)